app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PiecesController < ApplicationController
+  def update
+    @piece = Pieces.find(params[:id])
+    new_x = params[:x].to_i
+    new_y = params[:y].to_i
+    return update_attributes(x_position: new_x, y_position: new_y) if @piece.move!(new_x, new_y)
```

Comment:
Sounds good, I was not sure if to put it in the controller or in the model. I have noticed some people add a line a code like `update_attributes`  in the controllers but it makes a lot of sense to make controllers control the requests and not make them do more. I have updated the PR, thank you Jacob.